### PR TITLE
Add handling for funding period events and oracle data

### PR DESCRIPTION
**Pull request by witgaw**
+ notebook for data download

### CODE REVIEW

File: vega_sim/api/data.py [hunk 1]
```diff
@@ -304,12 +305,26 @@ class LiquidityProviderFeeShare:
     average_score: float
 
 
+@dataclass(frozen=True)
+class LiquiditySLA:
```
Comment (cdummett) [inline vega_sim/api/data.py]: Small nitpick but could we have this dataclass match the corresponding proto `LiquidityProviderSLA`?

File: vega_sim/api/data.py [hunk 2]
```diff
@@ -282,8 +282,9 @@ class MarketData:
     market_value_proxy: float
     price_monitoring_bounds: list
     liquidity_provider_fee_share: list
+    liquidity_sla: list
```
Comment (cdummett) [inline vega_sim/api/data.py]: Would give some useful type hinting when working with the `MarketData` dataclass.
```suggestion
    liquidity_sla: List[LiquidityProviderSLA]
```

File: vega_sim/api/data.py [hunk 3]
```diff
@@ -1053,7 +1123,7 @@ def positions_by_market(
     asset_decimals_map: Optional[Dict[str, int]] = None,
 ) -> Union[Dict[str, Position], Position]:
     """Output positions of a party."""
-
+    data_raw
```
Comment (cdummett) [inline vega_sim/api/data.py]:
```suggestion
```

File: vega_sim/api/data.py [hunk 4]
```diff
@@ -1245,6 +1334,73 @@ def _liquidity_provider_fee_share_from_proto(
...
+                for x in individual_liquidity_provider_sla.hysteresis_period_fee_penalties
+            ],
+            required_liquidity=(
+                0
+                if individual_liquidity_provider_sla.required_liquidity == ""
+                else num_from_padded_int(
+                    float(individual_liquidity_provider_sla.required_liquidity),
+                    decimals,
+                )
+            ),
```
Comment (cdummett) [inline vega_sim/api/data.py:1381]: I'm not 100% but am trying to check now. Are the right decimals being used here?

I would have assumed it was just the asset decimals.

File: vega_sim/api/data.py (hunk 1 above)
Comment (witgaw) [reply vega_sim/api/data.py]: Renamed now

File: vega_sim/api/data.py (hunk 4 above)
Comment (witgaw) [reply vega_sim/api/data.py:1381]: Yeah, after a bit of digging it does seem to be asset decimals, changed now.

**Review by cdummett — APPROVED** [review]

**Pull request merged**